### PR TITLE
Fix invalid boilerplate script issue

### LINKED ISSUE #4038: Fix invalid boilerplate
/kind cleanup

**What needs cleanup:**

`make verify-boilerplate` has been failing for some time because the script that invokes doesn't fail when the `python` command isn't available, which it's not in the prow environment:

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/directory/pull-cluster-api-provider-azure-verify/1706387974339432448#1:build-log.txt%3A21

**Describe the solution you'd like**
The script should be updated to invoke `python3` instead of `python` like CAPI: https://github.com/kubernetes-sigs/cluster-api/blob/4c6e89cf2ca962599856ff8e302894bdd4387bdf/hack/boilerplate/boilerplate.py#L1.

Once that change is made, `make verify-boilerplate` will report errors that should also be fixed.

**Anything else you would like to add:**

### PR DESCRIPTION
**Pull request by wowsoundaryaa**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

This PR should fix boilerplate script failing when `python` version is not available.
Changes includes
- change command `python` to `python3`
- update dockerfile ref file with correct version
- Fix space related issues in copyright headers

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #4038

### CODE REVIEW

**Comment by nojnhuh** [comment]
/lgtm
/release-note-none

Thank you @wowsoundaryaa! Could you please squash all commits into one and then force-push to this branch?
/hold

**Comment by wowsoundaryaa** [comment]
@nojnhuh squashed the commits. Thanks

**Comment by nojnhuh** [comment]
Thank you for squashing!
/hold cancel

File: azure/errors_test.go [hunk 1]
```diff
@@ -1,3 +1,19 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (Jont828) [inline azure/errors_test.go:2]:
```suggestion
Copyright 2023 The Kubernetes Authors.
```

**Review by Jont828 — COMMENTED** [review]
LGTM aside from a nit

File: azure/errors_test.go (hunk 1 above)
Comment (nojnhuh) [reply azure/errors_test.go:2]: As best as I can tell, this file was originally added in 2019. Not sure exactly what it means that there wasn't an explicit copyright on it before then, but I think it's fine as-is.

Comment (Jont828) [reply azure/errors_test.go:2]: Gotcha, I just wasn't sure if the copyright header is allowed to be retroactive or not

**Pull request merged**